rcl/src/rcl/init.c

```diff
@@ -144,6 +148,60 @@ rcl_init(
   rcutils_atomic_store((atomic_uint_least64_t *)(&context->instance_id_storage), next_instance_id);
   context->impl->init_options.impl->rmw_init_options.instance_id = next_instance_id;
 
+  // Get actual domain id based on environment variable, if needed.
+  ret = rcl_domain_id(&context->impl->init_options.impl->rmw_init_options.domain_id);
+  if (RCL_RET_OK != ret) {
+    fail_ret = ret;
+    goto fail;
+  }
+
+  // Get actual localhost_only value based on environment variable, if needed.
+  ret = rcl_localhost_only(&context->impl->init_options.impl->rmw_init_options.localhost_only);
+  if (RCL_RET_OK != ret) {
+    fail_ret = ret;
+    goto fail;
+  }
+
+  // TODO(ivanpauno): Uncomment this when new path discovery magic is ready.
+  bool use_security = false;
+  ret = rcl_use_security(&use_security);
+  if (RCL_RET_OK != ret) {
+    fail_ret = ret;
+    goto fail;
+  }
+  RCUTILS_LOG_DEBUG_NAMED(
+    ROS_PACKAGE_NAME, "Using security: %s", use_security ? "true" : "false");
+
+  if (!rmw_use_node_name_in_security_directory_lookup()) {
```

Comment:
So if the RMW sets this to be true, this chunk of code doesn't do anything and the old behavior in rcl_node_init() takes effect, right? It makes for what looks like quite a bit of duplication between these two functions-- is there a reason they can't be closer to each other? Or at least could we refactor this process into a single function call that does the right thing depending on RMW support?

Reply:
Yes, I think I can avoid most of the duplication.

Reply:
Done in https://github.com/ros2/rcl/pull/515/commits/b6ed0b46706415ea063b697da30f75cb8c9749fa.